src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -296,27 +314,37 @@ protected virtual Task<bool> HandleUnauthorizedAsync(ChallengeContext context)
                 {
                     case ChallengeBehavior.Automatic:
                         // If there is a principal already, invoke the forbidden code path
-                        var ticket = await HandleAuthenticateOnceAsync();
-                        if (ticket?.Principal != null)
+                        var result = await HandleAuthenticateOnceAsync();
+
+                        //if (result?.Error != null)
+                        //{
+                        //    await HandleErrorAsync(result.Error);
+                        //    if (result.Error.IsRequestComplete)
+                        //    {
+                        //        context.CompleteRequest();
+                        //    }
+                        //}
+
+                        if (result?.Ticket?.Principal != null)
                         {
-                            handled = await HandleForbiddenAsync(context);
+                            await HandleForbiddenAsync(context);
                         }
                         else
                         {
-                            handled = await HandleUnauthorizedAsync(context);
+                            await HandleUnauthorizedAsync(context);
                         }
                         break;
                     case ChallengeBehavior.Unauthorized:
-                        handled = await HandleUnauthorizedAsync(context);
+                        await HandleUnauthorizedAsync(context);
                         break;
                     case ChallengeBehavior.Forbidden:
-                        handled = await HandleForbiddenAsync(context);
+                        await HandleForbiddenAsync(context);
                         break;
                 }
                 context.Accept();
             }
 
-            if (!handled && PriorHandler != null)
+            if (PriorHandler != null)
```

Comment:
So there's no longer a mechanism to end chaining? Was that OK because any conflicting chaining would be the result of a configuration error? e.g. Basic and Cookies being set to automatic?